src/instrument/mongoose_instrument_exometer.erl

```diff
@@ -166,10 +170,17 @@ exometer_labels(#{pool_tag := PoolTag}) ->
 exometer_labels(#{}) ->
     [].
 
--spec get_host_type_prefix(mongoose_instrument:labels()) -> mongooseim:host_type_or_global().
+-spec get_host_type_prefix(atom() | binary() | mongoose_instrument:labels()) ->
+    mongooseim:host_type_or_global().
+get_host_type_prefix(HostType) when is_binary(HostType) ->
+    get_host_type_prefix(#{host_type => HostType});
+get_host_type_prefix(global) ->
+    global;
 get_host_type_prefix(#{host_type := HostType}) ->
-    #{HostType := Prefix} = persistent_term:get(?PREFIXES),
-    Prefix;
+    case persistent_term:get(?PREFIXES, #{}) of
+        #{HostType := Prefix} -> Prefix;
+        #{} -> make_host_type_prefix(HostType, all_metrics_are_global())
```

Comment:
This part of the code is necessary because someone might provide an incorrect host type when using the API. The existing logic first creates the prefix and then checks if the metrics exist.

Reply:
But if someone provides an incorrect host type, then there is no need to do this, because the metric will not exist anyway, right? I guess the only time it would return a correct metric anyway, would be when indeed `all_metrics_are_global`. But why is somebody checking a metric for a host type then? They would get a global value anyway. I'm not sure if somehow failing isn't better in this case.

Reply:
Question 1: Yes, but to check if the metric exists, we first need to get a prefix and then call the function in exometer. Alternatively, we could check if the prefix exists, and if it doesn't, simply return an empty list, skipping the call to exometer.

Question 2: I agree with you that it would be more reasonable to fail when checking a value for a given host type if `all_metrics_are_global` is true. But, that’s how the API using `mongoose_metric` was designed, and I think it’s a good idea to maintain this behavior in case some implementations rely on it.

Reply:
OK, I see this behaviour was the same in the old implementation. It's confusing for me when reading the code, and I'd rather change it, but I see why you don't want to change it in this PR.